src/ui/app/jsx/repair-schedule-form.jsx

```diff
@@ -0,0 +1,274 @@
+      owner: this.props.repair ? this.props.repair.owner : '',
+      startTime: this.props.repair ? this.props.repair.next_activation : (this.props.formType === 'schedule' ? moment().toDate() : null),
+      intervalDays: this.props.repair ? this.props.repair.scheduled_days_between : '',
+      tables: this.props.repair ? this.props.repair.column_families : [],
+      tablesBlacklisted: this.props.repair ? this.props.repair.blacklisted_tables : [],
+      nodes: this.props.repair ? this.props.repair.nodes : [],
+      datacenters: this.props.repair ? this.props.repair.datacenters : [],
+      segments: this.props.repair ? this.props.repair.segment_count_per_node : '',
+      parallelism: this.props.repair ? this.props.repair.repair_parallelism : 'PARALLEL',
+      intensity: this.props.repair ? this.props.repair.intensity : '',
+      incrementalRepair: this.props.repair ? this.props.repair.incremental_repair : 'false',
+      repairThreadCount: this.props.repair ? this.props.repair.repair_thread_count : 1,
+      formType: this.props.formType,
+      advancedSettingsOpen: false,
+      valid: undefined  // Indicates if the current state of the form is valid or not
+    }
+  },
+
+  _formSelectChangeHandler: function(valueContext, actionContext) {
+    const field = actionContext.name.split("_")[1];
+    const value = valueContext.value;
+    this._setStateFormChange(field, value);
+  },
+  
+  _formInputChangeHandler: function(e) {
+    const field = e.target.id.split("_")[1];
+    const value = e.target.value;
+    this._setStateFormChange(field, value);
+  },
+
+  _setStateFormChange: function(field, value) {
+    if (!field || !field.length) {
+      return;
+    }
+
+    const changeEvent = {
+      field: field,
+      value: value ? value.toString() : undefined,
+      origValue: this.state && field && this.state[field] ? this.state[field].toString() : undefined
+    }
+
+    const state = this.state;
+    if (field) {
+      state[field] = value;
+    }
+    state.valid = this._validate(state);
+    this.setState(state);
+
+    changeEvent.state = state;
+    if (this.props.onChange) {
+      this.props.onChange(changeEvent);
+    }
+  },
+
+  _validate: function(state) {
+    if (!state) {
+      return false;
+    }
+
+    const idInvalid = !state.id || !state.id.length;
+    const ownerInvalid = !state.owner || !state.owner.length;
+    const scheduleTimeInvalid = (state.formType === "schedule" &&
+        (!state.startTime ||
+          state.intervalDays == undefined ||
+          state.intervalDays == null ||
+          state.intervalDays < 0 ||
+          state.intervalDays > 31
+        )
+      ) ||
+      (state.formType === "repair" &&
+        (state.startTime ||
+          state.intervalDays
+        )
+      );
+    const segmentsPerNodeInvalid = !state.segments ||
+      state.segments == undefined ||
+      state.segments == null ||
+      state.segments < 1 ||
```

Comment:
We can relax this constraint as mentioned in my previous comment and allow 0 as segment count per node.